CalibMuon/CSCCalibration/plugins/CSCBadChambersConditions.cc

```diff
@@ -35,7 +35,7 @@ CSCBadChambersConditions::ReturnType
 CSCBadChambersConditions::produceBadChambers(const CSCBadChambersRcd& iRecord)
 {
   //need a new object so to not be deleted at exit
-  CSCBadChambers* mydata=new CSCBadChambers( *cndbBadChambers );
+  CSCBadChambersConditions::ReturnType mydata = std::make_unique<CSCBadChambers>( *cndbBadChambers );
```

Comment:
I think it would be cleaner to write something like this:
```
auto mydata = std::make_unique<CSCBadChambersConditions::ReturnType::element_type>(*cndbBadChambers);
```
That way you also get the possible constness of the pointed-to type correct.

Reply:
I definitely agree that auto would be better here

Reply:
Actually, I think this code could be significantly simplified by
   1. removing the member data `cndbBadChambers`
   1. removing the call to ` prefillBadChambers` in the constructor
   1. changing `produceBadChambers` to just be the following

```
return cndbBadChambers()
```

This also better aligns with the framework expectation that the module does not do work unless its data is requested.